pkg/cluster/eks/action/actions.go

```diff
@@ -1551,34 +1551,54 @@ func (a *WaitResourceDeletionAction) waitUntilELBsDeleted() error {
 
 	for {
 
+		var loadBalancerNames []*string
 		describeLoadBalancers := &elb.DescribeLoadBalancersInput{}
-		loadBalancers, err := elbService.DescribeLoadBalancers(describeLoadBalancers)
-		if err != nil {
-			return err
-		}
 
-		var loadBalancerNames []*string
-		for _, description := range loadBalancers.LoadBalancerDescriptions {
-			loadBalancerNames = append(loadBalancerNames, description.LoadBalancerName)
+		for nextPage := true; nextPage; nextPage = describeLoadBalancers.Marker != nil {
+			loadBalancers, err := elbService.DescribeLoadBalancers(describeLoadBalancers)
+			if err != nil {
+				return errors.WrapIf(err, "couldn't describe ELBs")
+			}
+
+			for _, description := range loadBalancers.LoadBalancerDescriptions {
+				loadBalancerNames = append(loadBalancerNames, description.LoadBalancerName)
+			}
+
+			describeLoadBalancers.Marker = loadBalancers.NextMarker
 		}
 
+
 		if len(loadBalancerNames) == 0 {
 			return nil
 		}
 
-		describeTagsInput := &elb.DescribeTagsInput{
-			LoadBalancerNames: loadBalancerNames,
-		}
-		describeTagsOutput, err := elbService.DescribeTags(describeTagsInput)
-		if err != nil {
-			return err
-		}
+		// according to https://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_DescribeTags.html
+		// tags can be queried for up to 20 ELBs in one call
 
 		var result []*string
-		for _, tagDescription := range describeTagsOutput.TagDescriptions {
-			for _, tag := range tagDescription.Tags {
-				if aws.StringValue(tag.Key) == clusterTag {
-					result = append(result, tagDescription.LoadBalancerName)
+		maxELBNames := 20
+		for i := 0; i < len(loadBalancerNames) / maxELBNames; i++ {
```

Comment:
last page is dropped if length%20 != 0.
```suggestion
		for low := 0; low < len(loadBalancerNames); i+= maxELBNames {
```